huawei/integrationpluginhuawei.cpp

```diff
@@ -0,0 +1,322 @@
+        return;
+    }
+
+    NetworkDeviceDiscoveryReply *discoveryReply = hardwareManager()->networkDeviceDiscovery()->discover();
+    connect(discoveryReply, &NetworkDeviceDiscoveryReply::finished, this, [=](){
+        foreach (const NetworkDeviceInfo &networkDeviceInfo, discoveryReply->networkDeviceInfos()) {
+
+            qCDebug(dcHuawei()) << "Found" << networkDeviceInfo;
+
+            // Filter for mac manufacturer
+            if (!networkDeviceInfo.macAddressManufacturer().contains("Huawei"))
+                continue;
+
+            QString title;
+            if (networkDeviceInfo.hostName().isEmpty()) {
+                title = "Huawei FusionSolar";
+            } else {
+                title = networkDeviceInfo.hostName() + " (" + networkDeviceInfo.address().toString() + ")";
+            }
+
+            QString description;
+            if (networkDeviceInfo.macAddressManufacturer().isEmpty()) {
+                description = networkDeviceInfo.macAddress();
+            } else {
+                description = networkDeviceInfo.macAddress() + " (" + networkDeviceInfo.macAddressManufacturer() + ")";
+            }
+
+            ThingDescriptor descriptor(huaweiInverterThingClassId, title, description);
+            ParamList params;
+            params << Param(huaweiInverterThingIpAddressParamTypeId, networkDeviceInfo.address().toString());
+            params << Param(huaweiInverterThingMacAddressParamTypeId, networkDeviceInfo.macAddress());
+            descriptor.setParams(params);
+
+            // Check if we already have set up this device
+            Things existingThings = myThings().filterByParam(huaweiInverterThingMacAddressParamTypeId, networkDeviceInfo.macAddress());
+            if (existingThings.count() == 1) {
+                qCDebug(dcHuawei()) << "This connection already exists in the system:" << networkDeviceInfo;
+                descriptor.setThingId(existingThings.first()->id());
+            }
+
+            info->addThingDescriptor(descriptor);
+        }
+
+        info->finish(Thing::ThingErrorNoError);
+    });
+}
+
+void IntegrationPluginHuawei::startMonitoringAutoThings()
+{
+
+}
+
+void IntegrationPluginHuawei::setupThing(ThingSetupInfo *info)
+{
+    Thing *thing = info->thing();
+    qCDebug(dcHuawei()) << "Setup" << thing << thing->params();
+
+    if (thing->thingClassId() == huaweiInverterThingClassId) {
+        QHostAddress hostAddress = QHostAddress(thing->paramValue(huaweiInverterThingIpAddressParamTypeId).toString());
+        if (hostAddress.isNull()) {
+            info->finish(Thing::ThingErrorInvalidParameter, QT_TR_NOOP("No IP address given"));
+            return;
+        }
+
+        uint port = thing->paramValue(huaweiInverterThingPortParamTypeId).toUInt();
+        quint16 slaveId = thing->paramValue(huaweiInverterThingSlaveIdParamTypeId).toUInt();
+
+        HuaweiFusionSolar *connection = new HuaweiFusionSolar(hostAddress, port, slaveId, this);
```

Comment:
This, again, seems to just break if some IP changes in the network, and it will.

Reply:
Will be added with the network device monitor